typedconfig/casts.py

```diff
@@ -0,0 +1,11 @@
+from enum import Enum, EnumMeta
+from typing import  Callable
+
+def enum_cast(enum_type: EnumMeta)->Callable[[str], Enum]:
+    def getter(name: str)->Enum:
+        if name not in enum_type._member_names_:
+            raise KeyError(f"{name} is not a member of {enum_type}")
+        cast_value = enum_type._member_map_[name]
+        return cast_value
+
```

Comment:
I believe this `getter` function as a whole can be written much more simply, avoiding using internal attributes of the `enum_type`. Still throws a `KeyError` if it's not valid - if you wanted to catch the `KeyError` and throw another `KeyError` with a better error message (such as the one you're using here) that's might be worth it though.
```python
def getter(name: str) -> Enum:
    return enum_type[name]
```

Reply:
Also some code style things here (e.g. having a space either side of  `->`). To make this consistent I've just added the `black` code style to the project (so you'll probably need to rebase this PR off master). Then you can autoformat the code with `black typedconfig test`.